spec/lib/common/client/middleware/request/remove_cookies_spec.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+describe Common::Client::Middleware::Request::RemoveCookies do
+  class TestConfiguration < Common::Client::Configuration::REST
+    def port
+      3010
+    end
+
+    def connection
+      @conn ||= Faraday.new("http://127.0.0.1:#{port}") do |faraday|
+        faraday.use(Common::Client::Middleware::Request::RemoveCookies)
+        faraday.adapter :httpclient
+      end
+    end
+  end
+
+  class TestService < Common::Client::Base
+    configuration TestConfiguration
+  end
+
+  describe '#request' do
+    let!(:server_thread) do
+      Thread.new do
```

Comment:
seems a little complicated, but this is a super complete way to test this functionality

Reply:
i was originally testing it with VCR but when the requests are mocked the cookie doesn't get set